### PR TITLE
feat: apply `ignorePresets` when resolving the `globalExtends` array

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Changes
Respect self hosted `ignorePresets` config option when resolving the `globalExtends` array.

## Context
- https://github.com/renovatebot/renovate/discussions/32596#discussioncomment-11318670

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/global/index.ts [hunk 1]
```diff
@@ -133,10 +134,13 @@ export async function start(): Promise<number> {
     await instrument('config', async () => {
       // read global config from file, env and cli args
       config = await getGlobalConfig();
-      if (config?.globalExtends) {
+      if (is.nonEmptyArray(config?.globalExtends)) {
```
Comment (viceice) [inline lib/workers/global/index.ts:137]: why this change?

Comment (Gabriel-Ladzaretti) [reply lib/workers/global/index.ts:137]: because for some reason it is an empty array even when not explicitly defined, this if condition is basically always truthy.
this has been my experience when debugging this section at least.

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]
